Enhance README (#21)
sayakpaul/deploy-hf-tf-vision-models: README.md

## Methods covered

- [x] Local TensorFlow Serving | [Blog post from 🤗](https://huggingface.co/blog/tf-serving-vision)

>   - We cover how to run a custom TensorFlow Serving with 🤗 Transformers' ViT model. 

Reviewer: ```suggestion
  - We cover how to locally deploy a Vision Transformer (ViT) model from 🤗 Transformers with TensorFlow Serving. 
```

  - With this, you will be able to serve and scale your own machine learning models according to the CPU utilizations of the deployment as a whole.
- [x] [Vertex AI Prediction](https://cloud.google.com/vertex-ai/docs/predictions/getting-predictions) 
  - We cover how to deploy 🤗 Transformers' ViT model to Google Cloud's fully managed machine learning deployment service (Vertex AI Prediction). 
  - Under the hood, Vertex AI Prediction leverages all the technologies from GKE, TensorFlow Serving, and more. 
  - That means you can deploy and scale the deployment of machine learning models, but you don't need to worry to make it happened from building a custom Docker image to writing Kubernetes specific menifests to setting up model monitoring capability.
  - With this, you will be able to serve and scale your own machine learning model by calling various APIs from `google-cloud-aiplatform` SDK to interact with Vertex AI. 

- [ ] Vertex AI Prediction (w/ [optimized TFRT](https://cloud.google.com/vertex-ai/docs/predictions/optimized-tensorflow-runtime))
  - TBD

> Know more about the optimized TFRT(TensorFlow RunTime) [here](https://github.com/tensorflow/runtime).

Reviewer: Do you mind including notes on Locust (GKE and Vertex AI) too?

Author: 👌

## Methods covered

- [x] [Local TensorFlow Serving](https://github.com/sayakpaul/deploy-hf-tf-vision-models/blob/main/hf_vision_model_tfserving.ipynb) | [Blog post from 🤗](https://huggingface.co/blog/tf-serving-vision)
  - We cover how to locally deploy a Vision Transformer (ViT) model from 🤗 Transformers with TensorFlow Serving. 
  - With this, you will be able to serve your own machine learning models in a standalone Python application.

> - [x] [TensorFlow Serving on Kubernetes(GKE)](https://github.com/sayakpaul/deploy-hf-tf-vision-models/tree/main/hf_vision_model_tfserving_gke) | [Blog post from 🤗](https://huggingface.co/blog/deploy-tfserving-kubernetes)

Reviewer: Since this one has a `locust` component too, should we move the current `locust` directory inside `hf_vision_model_tfserving_gke` and call it out from this section? 

WDYT?

Author: That makes sense!

  - We cover how to build a custom TensorFlow Serving Docker image with 🤗 Transformers' ViT model, provision GKE cluster, deploy the Docker image to the GKE cluster.
  - Particularly, we cover Kubernetes specific topics such as creating Deployment/Service/HPA Kubernetes objects for scalable deployment of the Docker image to the nodes(VMs) and expose them as a service to clients.
  - With this, you will be able to serve and scale your own machine learning models according to the CPU utilizations of the deployment as a whole.
- [x] [Vertex AI Prediction](https://cloud.google.com/vertex-ai/docs/predictions/getting-predictions) 
  - We cover how to deploy 🤗 Transformers' ViT model to Google Cloud's fully managed machine learning deployment service (Vertex AI Prediction). 
  - Under the hood, Vertex AI Prediction leverages all the technologies from GKE, TensorFlow Serving, and more. 

>   - That means you can deploy and scale the deployment of machine learning models, but you don't need to worry to make it happened from building a custom Docker image to writing Kubernetes specific menifests to setting up model monitoring capability.

Reviewer: ```suggestion
  - That means you can deploy and scale the deployment of machine learning models, but you don't need to worry about building a custom Docker image or writing Kubernetes-specific manifests, or setting up model monitoring capability.
```

  - With this, you will be able to serve and scale your own machine learning model by calling various APIs from `google-cloud-aiplatform` SDK to interact with Vertex AI. 

>   - Also, we provide extra codes to show how to perform load-test with [Locust](https://locust.io/), and you will find how to construct requests and send them to the Vertex AI Endpoint without `PredictionServiceClient` module dependency from `google-cloud-aiplatform` SDK.

Reviewer: ```suggestion
  - We provide utilities to perform load-test with [Locust](https://locust.io/). Refer [here](./hf_vision_model_vertex_ai/locust) for more details.
```
